AMM/project/scripts/atomic-add-liquidity/src/main.sw

```diff
@@ -0,0 +1,44 @@
+script;
+
+use libraries::Exchange;
+use std::block::height;
+
+enum InputError {
+    DesiredLiquidityZero: (),
+}
+
+fn main(
+    exchange_contract_id: ContractId,
+    assets: (ContractId, ContractId),
+    deposit_amounts: (u64, u64),
+    desired_liquidity: u64,
+) -> u64 {
+    require(desired_liquidity > 0, InputError::DesiredLiquidityZero);
+
+    let exchange_contract = abi(Exchange, exchange_contract_id.into());
+
+    // deposit first asset
+    if deposit_amounts.0 != 0 {
+        exchange_contract.deposit {
+            gas: 100_000,
```

Comment:
Why should I trust this gas value (and others)? The SDK has a way to check the gas; however, that might change in the future. Can we avoid hardcoding gas at all?

Reply:
There is an issue with the estimate gas function but I applied your suggestion to the best I could. We can have further discussion on this.